BinaryEncoding.md

```diff
@@ -234,6 +233,52 @@ This section must be preceded by a [Functions](#functions-section) section.
 | count | `varuint32` | count of entries to follow |
 | entries | `uint16*` | repeated indexes into the function table |
 
+### Function Names section
+
+This section may occur 0 or 1 times and does not change execution semantics. A
+validation error in this section is not reported and is treated as the section
+being absent. The expectation is that, when a binary WebAssembly module is
+viewed in a browser or other development environment, the names in this section
+will be used as the names of functions in the [text format](TextFormat.md).
+
+ID: `function_names`
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| names | `name*` | sequence of names |
+
+The number of names is determined by the number of declared functions. The
+sequence of names assigns a name to each function index. A name is:
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| size | varuint32 | string length, in bytes |
+| string | bytes | valid utf8 encoding |
```

Comment:
@rossberg-chromium Hehe, yeah, I guess I should've said latin1.

What do you think about the current text above which says "you don't have to validate, but if you do at all, it has to be real utf8".  I think that models what we'd do in FF: if present (and maybe only if a devtools flag is set) decode as utf8.  If that fails, you don't get symbols and maybe we print a warning to the console saying invalid names section.